### PR TITLE
byebye torch 2.4

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

bye

### CODE REVIEW

**Review by vasqu — APPROVED** [review]
I think we have some additional stuff here
- https://github.com/huggingface/transformers/blob/d0e91c7ff1d3ab41b49de369a396f65191e34d2b/src/transformers/masking_utils.py#L37
- https://github.com/huggingface/transformers/blob/d0e91c7ff1d3ab41b49de369a396f65191e34d2b/src/transformers/integrations/sdpa_attention.py#L10

But otherwise looks good I think

File: src/transformers/masking_utils.py [hunk 1]
```diff
@@ -413,9 +411,6 @@ def sdpa_mask(
         allow_is_bidirectional_skip (`bool`, optional):
             Whether to allow to return `None` for the mask under conditions where we do not have to add any bias,
             i.e. full attention without any padding. Default to `False`.
-        allow_torch_fix (`bool`, optional):
-            Whether to update the mask in case a query is not attending to any tokens, to solve a bug in torch's older
-            versions. We need an arg to skip it when using eager. By default `True`.
```
Comment (ydshieh) [inline src/transformers/masking_utils.py:418]: I think we need to keep this argument, but not doing anything. The docstring needs to be updated.

File: src/transformers/masking_utils.py [hunk 2]
```diff
@@ -598,7 +587,6 @@ def eager_mask(
         attention_mask=attention_mask,
         allow_is_causal_skip=False,
         allow_is_bidirectional_skip=allow_is_bidirectional_skip,
-        allow_torch_fix=False,
```
Comment (ydshieh) [inline src/transformers/masking_utils.py:601]: we don't need to pass it now. The default is True, but it won't have any effect.

File: src/transformers/masking_utils.py (hunk 1 above)
Comment (vasqu) [reply src/transformers/masking_utils.py:418]: Can you use deprecate kwarg decorator for now then?

Comment (ydshieh) [reply src/transformers/masking_utils.py:418]: or if it's really only internal and we want to make it clean, I am fine to remove it.

File: src/transformers/masking_utils.py [hunk 3]
```diff
@@ -532,11 +532,6 @@ def sdpa_mask(
             "Please update your torch version or use `use_vmap=False` with index-based masks."
         )
 
-    # Due to a bug in versions of torch<2.5, we need to update the mask in case a query is not attending to any
-    # tokens (due to padding). See details in https://github.com/pytorch/pytorch/issues/110213
-    if not _is_torch_greater_or_equal_than_2_5 and allow_torch_fix:
-        attention_mask = attention_mask | torch.all(~attention_mask, dim=-1, keepdim=True)
```
Comment (Cyrilvallez) [inline src/transformers/masking_utils.py:538]: So happy to remove this from the masks finally 🙏🥲

**Review by Cyrilvallez — APPROVED** [review]
Let's go!
